### PR TITLE
Vertx 3.5.1

### PR DESCRIPTION
**Pull request by pratikpparikh**
Dear Contributor,

updated to latest vertx version.

Regards,
Pratik Parikh

### CODE REVIEW

**Comment by pratikpparikh** [comment]
@msavy what are the next steps for this pull request?

File: gateway/platforms/vertx3/vertx3/src/main/java/io/apiman/gateway/platforms/vertx3/api/Router2ResteasyRequestAdapter.java [hunk 1]
```diff
@@ -111,6 +112,11 @@ public String rawMethod() {
         return request.peerCertificateChain();
     }
 
+    @Override
+    public SSLSession sslSession(){
+        return null;
```
Comment (msavy) [inline gateway/platforms/vertx3/vertx3/src/main/java/io/apiman/gateway/platforms/vertx3/api/Router2ResteasyRequestAdapter.java]: Is there any reason for this being null?

File: gateway/platforms/vertx3/vertx3/src/main/java/io/apiman/gateway/platforms/vertx3/api/auth/KeycloakOAuthFactory.java [hunk 2]
```diff
@@ -140,6 +143,14 @@ public AuthHandler addAuthority(String authority) {
             public AuthHandler addAuthorities(Set<String> authorities) {
                 return this;
             }
+
+            public void parseCredentials(RoutingContext context, Handler<AsyncResult<JsonObject>> handler){
+                // do nothing
```
Comment (msavy) [inline gateway/platforms/vertx3/vertx3/src/main/java/io/apiman/gateway/platforms/vertx3/api/auth/KeycloakOAuthFactory.java]: What's the rationale for doing nothing?

File: gateway/platforms/vertx3/vertx3/src/main/java/io/apiman/gateway/platforms/vertx3/api/auth/KeycloakOAuthFactory.java [hunk 3]
```diff
@@ -140,6 +143,14 @@ public AuthHandler addAuthority(String authority) {
...
+            }
+
+            public void authorize(User user, Handler<AsyncResult<Void>> handler){
+                // do nothing
```
Comment (msavy) [inline gateway/platforms/vertx3/vertx3/src/main/java/io/apiman/gateway/platforms/vertx3/api/auth/KeycloakOAuthFactory.java]: As above

**Comment by msavy** [comment]
Do you have any particular reason for needing Vert.x 3.5.1?

**Comment by msavy** [comment]
No response from OP. Closing. Feel free to reopen if you want to pursue this further.

**Pull request closed without merging**